posthog/models/filters/mixins/simplify.py

```diff
@@ -25,21 +26,34 @@ def simplify(self: T, team: "Team", **kwargs) -> T:
 
         # :TRICKY: Make a copy to avoid caching issues
         result: Any = self.with_data({"is_simplified": True})  # type: ignore
+
         if getattr(result, "filter_test_accounts", False):
-            result = result.with_data(
-                {"properties": result.properties + team.test_account_filters, "filter_test_accounts": False,}
+
+            new_group = {"type": "AND", "values": team.test_account_filters}
+            prop_group = (
+                {"type": "AND", "values": [new_group, result.property_groups.to_dict()]}
+                if result.property_groups.to_dict()
+                else new_group
             )
+            result = result.with_data({"properties": prop_group, "filter_test_accounts": False,})
```

Comment:
Bit of nesting that will occur now because of the group structure. If test accounts exists and group filtering exists, they will be nested rather than having a simple array append